SU2_SOL/src/SU2_SOL.cpp

```diff
@@ -105,49 +108,70 @@ int main(int argc, char *argv[]) {
     config_container[iZone] = new CConfig(config_file_name, SU2_SOL, iZone, nZone, 0, VERB_HIGH);
     config_container[iZone]->SetMPICommunicator(MPICommunicator);
 
-    /*--- Definition of the geometry class to store the primal grid in the partitioning process. ---*/
+    /*--- Read the number of instances for each zone ---*/
 
-    CGeometry *geometry_aux = NULL;
+    nInst[iZone] = config_container[iZone]->GetnTimeInstances();
 
-    /*--- All ranks process the grid and call ParMETIS for partitioning ---*/
+    geometry_container[iZone] = new CGeometry*[nInst[iZone]];
+    solver_container[iZone] = new CSolver*[nInst[iZone]];
 
-    geometry_aux = new CPhysicalGeometry(config_container[iZone], iZone, nZone);
+    for (iInst = 0; iInst < nInst[iZone]; iInst++){
 
-    /*--- Color the initial grid and set the send-receive domains (ParMETIS) ---*/
+      /*--- Allocate solver. ---*/
+      solver_container[iZone][iInst] = NULL;
 
-    geometry_aux->SetColorGrid_Parallel(config_container[iZone]);
+      config_container[iZone]->SetiInst(iInst);
 
-    /*--- Until we finish the new periodic BC implementation, use the old
-     partitioning routines for cases with periodic BCs. The old routines 
-     will be entirely removed eventually in favor of the new methods. ---*/
+      /*--- Definition of the geometry class to store the primal grid in the partitioning process. ---*/
 
-    if (periodic) {
-      geometry_container[iZone] = new CPhysicalGeometry(geometry_aux, config_container[iZone]);
-    } else {
-      geometry_container[iZone] = new CPhysicalGeometry(geometry_aux, config_container[iZone], periodic);
-    }
+      CGeometry *geometry_aux = NULL;
+
+      /*--- All ranks process the grid and call ParMETIS for partitioning ---*/
+
+      geometry_aux = new CPhysicalGeometry(config_container[iZone], iZone, nZone);
+
+      /*--- Color the initial grid and set the send-receive domains (ParMETIS) ---*/
+
+      geometry_aux->SetColorGrid_Parallel(config_container[iZone]);
+
+      /*--- Allocate the memory of the current domain, and
+     divide the grid between the nodes ---*/
+
+      geometry_container[iZone][iInst] = NULL;
+
+      /*--- Until we finish the new periodic BC implementation, use the old
+       partitioning routines for cases with periodic BCs. The old routines 
+       will be entirely removed eventually in favor of the new methods. ---*/
 
-    /*--- Deallocate the memory of geometry_aux ---*/
+      if (periodic) {
+        geometry_container[iZone][iInst] = new CPhysicalGeometry(geometry_aux, config_container[iZone]);
+      } else {
+        geometry_container[iZone][iInst] = new CPhysicalGeometry(geometry_aux, config_container[iZone], periodic);
+      }
+
+      /*--- Deallocate the memory of geometry_aux ---*/
 
-    delete geometry_aux;
+      delete geometry_aux;
```

Comment:
a few extraneous white space changes here. 
See below for tips on isolating white space changes quickly:
https://stackoverflow.com/questions/3515597/add-only-non-whitespace-changes/7149602#7149602

Reply:
The whitespace changes here are due to the new loop on iInst, which increases the indentation one level (or 2 spaces).